docs/_future/create_new_app.md

```diff
@@ -0,0 +1,765 @@
+
+In the Request Time Off application, the Time Off Request workflow is defined in `manifest/workflows/time-off-request.js`. The general structure of a workflow is to first define it using the built in function `DefineWorkflow`:
+
+```js
+const { DefineWorkflow, Schema } = require('@slack/bolt');
+
+const TimeOffWorkflow = DefineWorkflow({
+  callback_id: 'time_off_request_wf',
+  title: 'Time Off Request Workflow',
+  input_parameters: {
+    properties: {
+      interactivity: {
+        type: Schema.slack.types.interactivity,
+      },
+    },
+    required: [],
+  },
+});
+
+module.exports = { TimeOffWorkflow };
+```
+
+In order to create a workflow that utilizes user input, such as one that uses a form, you'll need to pass in `interactivity` as one of the workflow's `input_parameters` under `properties`. This will allow the form to be interacted with.
+
+Now that the workflow has been defined, you can start adding steps underneath the definition, which will execute additional functionality in your workflow. Steps are executed in the order they appear. For the Time Off Request workflow, the first step will be a form that will allow the user to enter in their time off request.
+
+```js
+const step1 = TimeOffWorkflow.addStep(
+  Schema.slack.functions.OpenForm,
+  {
+    title: 'Request Time Off',
+    submit_label: 'Request',
+    description: 'Please describe your request',
+    interactivity: TimeOffWorkflow.inputs.interactivity,
+    fields: {
+      required: ['manager', 'start_date', 'end_date'],
+      elements: [
+        {
+          name: 'manager',
+          title: 'Manager',
+          type: Schema.slack.types.user_id,
+        },
+        {
+          name: 'start_date',
+          title: 'Start Date',
+          type: Schema.slack.types.timestamp,
+        },
+        {
+          name: 'end_date',
+          title: 'End Date',
+          type: Schema.slack.types.timestamp,
+        },
+      ],
+    },
+  },
+);
+```
+This step uses the built-in function [OpenForm](https://api.slack.com/future/functions#open-a-form) and initializes expected inputs of the form, which are defined by [Slack types](https://api.slack.com/future/types), such as the `timestamp` type, which represents dates and times. The Time Off Request form contains a `manager`, `start_date`, and `end_date` field, all of which are required inputs to submit the form.
+
+After the form is submitted, another workflow step can be added to handle the information that was submitted in this form. For the Time Off Request workflow, after the requester submits the request, it will send a message to the manager to notify them of the request and allow them to approve or deny the request through the message. Since this functionality does not exist as a built-in function, it requires a custom function in order to execute the desired functionality. In our workflow, we'll call it `ApprovalFunction`:
```

Comment:
```suggestion
After the form is submitted, another workflow step handles the information that was submitted in this form. In our example, after the requester submits, we want to send a message to their manager to notify them of the request and allow them to approve or deny the request through the message. Since this functionality does not exist as a built-in function, let's write our own custom function to do this work. In our workflow, we'll call it `ApprovalFunction`:
```